From 12a643089184bff567dc3fe91c8cc01dfcf170a7 Mon Sep 17 00:00:00 2001
From: Arno Renevier <arno@renevier.net>
Date: Wed, 2 Apr 2025 10:05:19 -0700
Subject: [PATCH] fixes build failure with gdk3
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

qiv fails to build with the following error:

event.c:1161:74: error: passing argument 1 of ‘gdk_device_manager_get_client_pointer’ from incompatible pointer type [-Wincompa
tible-pointer-types]
 1161 |               GdkDevice* pointer = gdk_device_manager_get_client_pointer(device_manager);
      |                                                                          ^~~~~~~~~~~~~~
      |                                                                          |
      |                                                                          GList * {aka struct _GList *}
---
 event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/event.c b/event.c
index baec737..a420f92 100644
--- a/event.c
+++ b/event.c
@@ -1157,7 +1157,7 @@ void qiv_handle_event(GdkEvent *ev, gpointer data)
               gdk_keyboard_grab(q->win, FALSE, CurrentTime);
 #else
               GdkDisplay* display = gdk_display_get_default();
-              GList* device_manager = gdk_display_get_device_manager(display);
+              GdkDeviceManager* device_manager = gdk_display_get_device_manager(display);
               GdkDevice* pointer = gdk_device_manager_get_client_pointer(device_manager);
               gdk_device_grab(pointer, q->win, GDK_OWNERSHIP_NONE, FALSE,
                   GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK | GDK_ENTER_NOTIFY_MASK |
-- 
2.49.0

